Cognite.Simulator.Utils/ModelLibraryBase.cs

```diff
@@ -1,35 +1,79 @@
-﻿using Cognite.Extractor.StateStorage;
-using Cognite.Extractor.Utils;
-using Cognite.Simulator.Extensions;
-using CogniteSdk;
-using Microsoft.Extensions.Logging;
-using System;
+﻿using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
-using System.Text;
-using Serilog.Context;
 using System.Threading;
 using System.Threading.Tasks;
-using System.IO;
+using System.Collections.Concurrent;
+
+using Microsoft.Extensions.Logging;
+using Serilog.Context;
+
+using Cognite.Extractor.StateStorage;
+using Cognite.Extractor.Utils;
+
+using CogniteSdk;
 using CogniteSdk.Alpha;
+using CogniteSdk.Resources;
+using CogniteSdk.Resources.Alpha;
 using Cognite.Extensions;
+using Cognite.Extractor.Common;
 
 namespace Cognite.Simulator.Utils
 {
     /// <summary>
-    /// Represents a local model file library. This is a <see cref="FileLibrary{T, U}"/> that
-    /// fetches simulator model files from CDF, save a local copy and process the model (extract information).
+    /// Represents a local model file library.
+    /// It fetches simulator model files from CDF, save a local copy and process the model (extract information).
     /// This library only keeps the latest version of a given model file
     /// </summary>
     /// <typeparam name="T">Type of the state object used in this library</typeparam>
     /// <typeparam name="U">Type of the data object used to serialize and deserialize state</typeparam>
     /// <typeparam name="V">Type of the model parsing information object</typeparam>
-    public abstract class ModelLibraryBase<T, U, V> : FileLibrary<T, U>, IModelProvider<T>
+    public abstract class ModelLibraryBase<T, U, V> : IModelProvider<T>
         where T : ModelStateBase
         where U : ModelStateBasePoco
         where V : ModelParsingInfo, new()
     {
         private readonly ILogger _logger;
+
+        /// <summary>
+        /// Dictionary holding the file states. The keys are the ids of the
+        /// CDF simulator model revisions and the values are the state objects of type <typeparamref name="T"/>
+        /// Should only be modified from a single thread to avoid multi-threading issues.
+        /// </summary>
+        public ConcurrentDictionary<string, T> State { get; private set; }
+
+        /// <summary>
+        /// Temporary model states that are used once and then deleted after the run.
+        /// We keep them in memory to avoid saving them to the state store and hence to avoid the multi-threading issues.
```

Comment:
```suggestion
        /// We keep them in memory to avoid saving them to the state store and hence to avoid the multi-threading issues.
        /// This state is used to store temporary model file paths, used only in case a run event is received before the model file has been downloaded. 
```